plugins/woocommerce/includes/admin/class-wc-admin-notices.php

```diff
@@ -213,13 +215,42 @@ private static function maybe_add_legacy_api_removal_notice() {
 	}
 
 	/**
-	 * Remove the admin notice about the removal of the Legacy REST API if the said API is disabled
-	 * or if the Legacy REST API extension is installed, and remove the notice about Legacy webhooks
-	 * if no such webhooks exist anymore or if the Legacy REST API extension is installed.
+	 * Add a notice about the incompatibility between the Legacy REST API and HPOS if both are active
+	 * (regardless of whether the LEgacy REST API is in core or in the dedicated plugin )
+	 */
+	private static function maybe_add_legacy_api_hpos_incompatibility_notice() {
+		if ( ! ( 'yes' === get_option( 'woocommerce_api_enabled' ) && 'yes' === get_option( CustomOrdersTableController::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION ) ) ) {
```

Comment:
We also don't need to show this notice when sync is enabled, wdyt? same for the static notice too